src/System Application/App/File System/src/Scenario/FileScenarioImpl.Codeunit.al

```diff
@@ -0,0 +1,328 @@
+                FileAccount := AllFileAccounts;
+                exit(true);
+            end;
+
+        exit(false);
+    end;
+
+    procedure SetFileAccount(Scenario: Enum "File Scenario"; FileAccount: Record "File Account")
+    var
+        FileScenario: Record "File Scenario";
+    begin
+        if not FileScenario.Get(Scenario) then begin
+            FileScenario.Scenario := Scenario;
+            FileScenario.Insert();
+        end;
+
+        FileScenario."Account Id" := FileAccount."Account Id";
+        FileScenario.Connector := FileAccount.Connector;
+
+        FileScenario.Modify();
+    end;
+
+    procedure UnassignScenario(Scenario: Enum "File Scenario")
+    var
+        FileScenario: Record "File Scenario";
+    begin
+        if FileScenario.Get(Scenario) then
+            FileScenario.Delete();
+    end;
+
+    /// <summary>
+    /// Get a list of entries, representing a tree structure with file accounts and the scenarios, assigned to each accout.
+    /// </summary>
+    /// <example>
+    /// Account sales@cronus.com has scenarios "Sales Quote" and "Sales Credit Memo" assigned.
+    /// Account purchase@cronus.com has scenarios "Purchase Quote" and "Purchase Invoice" assigned.
+    /// The result of calling the function will be:
+    /// sales@cronus.com, "Sales Quote", "Sales Credit Memo", purchase@cronus.com, "Purchase Quote", "Purchase Invoice"
+    /// </example>
+    /// <param name="Result">A flatten tree structure representing the all the file accounts and the scenarios assigned to them.</param>
+    procedure GetScenariosByFileAccount(var Result: Record "File Account Scenario")
+    var
+        FileAccounts: Record "File Account";
+        FileAccountScenarios: Record "File Account Scenario";
+        DefaultAccount: Record "File Account";
+        FileAccount: Codeunit "File Account";
+        DisplayName: Text[2048];
+        Position: Integer;
+        Default: Boolean;
+    begin
+        Result.Reset();
+        Result.DeleteAll();
+
+        FileAccount.GetAllAccounts(FileAccounts);
+
+        if not FileAccounts.FindSet() then
+            exit; // No accounts, nothing to do
+
+        // The position is set in order to be able to properly sort the entries (by order of insertion)
+        Position := 1;
+        GetDefaultAccount(DefaultAccount);
+
+        repeat
+            Default := (FileAccounts."Account Id" = DefaultAccount."Account Id") and (FileAccounts.Connector = DefaultAccount.Connector);
+            DisplayName := FileAccounts.Name;
+
+            // Add entry for the file account. Scenario is -1, because it isn't needed when displaying the file account.
+            AddEntry(Result, Result.EntryType::Account, -1, FileAccounts."Account Id", FileAccounts.Connector, DisplayName, Default, Position);
+
+            // Get the file scenarios assigned to the current file account, sorted by "Display Name"
+            GetFileScenariosForAccount(FileAccounts, FileAccountScenarios);
```

Comment:
Double FindSet:
![image](https://github.com/user-attachments/assets/e4526bb1-9f81-457c-9ff0-dfaed1ba0aa5)

FileAccountScenarios are deleted in procedure GetFileScenariosForAccount

Meaning
if FileAccountScenarios.FindSet() then
Will always be false when executed.

Suggestion is to make procedure GetFileScenariosForAccount return success so you would exit above without triggering additional query to SQL

Reply:
The two findsets are not executed on the same instance. The Table is temporrary, SQL is also not envolved.

Reply:
Sorry, I missed it due to being on table level. Let it be like that, all good. Thanks!